### PR TITLE
DeviceOS Release 1.5.3

### PR DESCRIPTION
**Pull request by avtolstoy**
### Description

1.5.2 + Asset Tracker support.

#### FEATURES

- [Gen 3] Add API to read OTP hardware version and model information
- [Asset Tracker] Alternate I2C interface on D8/D9

#### ENHANCEMENTS

- [Asset Tracker] Improve external RTC accuracy

#### BUGFIXES

- [Asset Tracker] Hardware flow control should not be disable for revision 0 devices
- [Asset Tracker] Suspend program/erase opertion when QSPI flash init gives timeout error

### Steps to Test

N/A

### Example App

N/A

### References

N/A

---

### Completeness

- [x] User is totes amazing for contributing!
- [ ] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: build/create_module.py [hunk 1]
```diff
@@ -230,7 +231,7 @@ def parse_dependency(dep):
         return ModuleDependency(func, version, index)
     return ModuleDependency()
 
-GEN3_PLATFORMS = [Platform.ARGON, Platform.BORON, Platform.XENON, Platform.ASOM, Platform.BSOM, Platform.B5SOM]
+GEN3_PLATFORMS = [Platform.ARGON, Platform.BORON, Platform.XENON, Platform.BSOM, Platform.B5SOM, Platform.TRACKER]
```
Comment (eberseth) [inline build/create_module.py:234]: ASOM not needed anymore?

Comment (avtolstoy) [reply build/create_module.py:234]: :+1: Rebase artifact.

Comment (avtolstoy) [reply build/create_module.py]: Fixed.

**Comment by avtolstoy** [comment]
@eberseth @YutingYou @XuGuohui For the purposes of review, it might also be useful to diff with `release/v1.5.x-tracker` (should be available now in the public repo). It should be possible to temporarily retarget this PR to that branch.

**Comment by eberseth** [comment]
@avtolstoy Compiled and installed on device.  It went into safe mode

**Comment by avtolstoy** [comment]
Release binaries built on CI: https://s3.amazonaws.com/binaries.particle.io/device-os-automated/binaries/particle_device-os%401.5.3%2B018bba1aaae7be2e87aef85cca0812f0abc70f77.zip

**Comment by XuGuohui** [comment]
Tests:
1. RGB persistent overriden: 💯 
2. `TEST=wiring/no_fixture`: 💯 
3. `TEST=wiring/sleep20`: 💯 
4. `TEST=wiring/no_fixture_ble`: 💯 
5. `TEST=wiring/no_fixture_spi`:
    >Assertion failed: (transferTime=10359) <= (expectedTime=10358), file tests/wiring/no_fixture_spi/spi.cpp, line 640.
Test SPI_20_SPI_Transfer_1024_Bytes_Per_DMA_Transmission_Locking failed.
6. `TEST=wiring/no_fixture_long_running`: 
    >Assertion failed: (Network.ready()=0) == (true=1), file tests/wiring/no_fixture_long_running/network.cpp, line 75.
Test NETWORK_01_LargePacketsDontCauseIssues_ResolveMtu failed.
    >Assertion failed: (HAL_PWM_Get_Frequency_Ext(pin)=500) != (500=500), file tests/wiring/no_fixture_long_running/pwm.cpp, line 102.
Test PWM_02_NoAnalogWriteWhenPinSelectedIsNotTimerChannel failed.
    >Assertion failed: (pinAvailable(pin)=1) == (0=0), file tests/wiring/no_fixture_long_running/pwm.cpp, line 113.
Test PWM_03_NoAnalogWriteWhenPinSelectedIsOutOfRange failed.
    >Assertion failed: (((int32_t)now_millis - (int32_t)last_millis) <= 1=0) == (true=1), file tests/wiring/no_fixture_long_running/ticks.cpp, line 86.
Assertion failed: (millis()-start=51017) >= (TWO_MINUTES=120000), file tests/wiring/no_fixture_long_running/ticks.cpp, line 151.
Test TICKS_02_millis_and_micros_along_with_high_priority_interrupts failed.
7. Time keeping: 💯

**Comment by avtolstoy** [comment]
@XuGuohui Thanks for looking into this. Not sure what happened with network tests (failed to connect to cellular network?), but the only thing in the tests listed we probably need to resolve is the PWM pin list for the tracker platform.

**Pull request closed without merging**